### ISSUE TITLE
[bug] Crash on launch

### ISSUE THREAD

**Issue by niedzielski** [patriciogonzalezvivo/glslViewer#289]
Thank you for making this cool tool! I'd love to try it but it seems to crash without any diagnostic output. I tried v2.1.2 on Debian GNU / Linux 11 (Bullseye).

```
$ glslViewer --verbose
┌──────────────────────────────────────────────────────────────────────────────────────────────┐
│>                                                                                             │
└──$
```

A window appears briefly but the terminal gets screwed up (requires a `reset`) and the program exits almost immediately.

**Comment by martinleopold** [comment]
Same here on macOS 12.6, freshly built (no errors) for the upcoming course. Terminal messed up after crash as well.

https://user-images.githubusercontent.com/1692826/195301579-1535f9ff-03c7-4ace-8cc4-cf44718fba79.mov

Note that directly loading a shader seems to work e.g. `glslViewer examples/2D/00_tests/test.frag`, but the zsh prompt is messed up afterwards.

Also noticed, `glslViewer --help` shows
```
zsh: segmentation fault  glslViewer --help
```

<img width="1022" alt="Screen Shot 2022-10-12 at 11 11 58" src="https://user-images.githubusercontent.com/1692826/195302011-3dd9c539-7a13-476c-839c-e39659a3b5a0.png">

**Comment by patriciogonzalezvivo** [comment]
Hi! good catch! I fixed the error in this one https://github.com/patriciogonzalezvivo/glslViewer/commit/15a3e957cfa59b8509eb88d7cada94610bfe8645 

Do you mind giving it a try?

Next I will see how to prevent ncurses to break the terminal on errors.

**Comment by patriciogonzalezvivo** [comment]
I temporally comment the color changing. https://github.com/patriciogonzalezvivo/glslViewer/commit/69219797a0f0ee83a2668cad8cdc98455348fa8f

I will look later on improving the entire ncurses solution... feels a bit glitchy at the moment

**Comment by martinleopold** [comment]
Thanks, confirming colors remain unchanged now!

**Comment by patriciogonzalezvivo** [comment]
I belive this one was solved on the latest version 3.0.4.

**Comment by niedzielski** [comment]
Seems to work for me. Thank you!
